### PR TITLE
get (individual) eenp stake

### PR DESCRIPTION
**Pull request by qinwei-gong**

### CODE REVIEW

File: cmd/thetacli/cmd/query/eenp_stake.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	if err != nil {
+		utils.Error("Failed to parse server response: %v\n%s\n", err, string(json))
+	}
+	fmt.Println(string(json))
+}
+
+func init() {
+	eenpStakeCmd.Flags().StringVar(&sourceFlag, "source", "", "Source of the stake")
+	eenpStakeCmd.Flags().StringVar(&holderFlag, "holder", "", "Holder of the stake")
+	eenpStakeCmd.Flags().BoolVar(&withdrawnOnlyFlag, "withdrawn_only", true, "Only want withdrawn stake")
```
Comment (weiribao) [inline cmd/thetacli/cmd/query/eenp_stake.go:51]: Should this flag default to false as that is the more common use case?

File: rpc/query.go [hunk 2]
```diff
@@ -777,6 +777,48 @@ func (t *ThetaRPCService) GetEenpByHeight(args *GetEenpByHeightArgs, result *Get
...
+		return err
+	}
+
+	db := deliveredView.GetDB()
+	height := uint64(args.Height)
+
+	var stake *core.Stake
+	blocks := t.chain.FindBlocksByHeight(height)
+	for _, b := range blocks {
+		stateRoot := b.StateHash
```
Comment (weiribao) [inline rpc/query.go:805]: We should check block.Status to find the best block. We had bug caused by similar code before in the case of chain having more than 1 branch. 

We can do something like this:
```
		var candidate *core.ExtendedBlock
		for _, b := range blocks {
			switch b.Status {
			case core.BlockStatusDirectlyFinalized, core.BlockStatusIndirectlyFinalized, core.BlockStatusTrusted:
				candidate = b
				break
			case core.BlockStatusCommitted, core.BlockStatusValid, core.BlockStatusPending:
				if candidate == nil || b.Status > candidate.Status {
					candidate = b
				}
			}
		}
		if candidate == nil {
			...
		}
```

(Probably this should be a method in core/chain.go)

Comment (qinwei-gong) [reply rpc/query.go:805]: i see. i actually copied this code from existing "GetEenpByHeight" (line 759). shall we change both places?

File: cmd/thetacli/cmd/query/eenp_stake.go (hunk 1 above)
Comment (jieyilong) [reply cmd/thetacli/cmd/query/eenp_stake.go:51]: yeah probably better default to false to be backward compatible.

File: rpc/query.go (hunk 2 above)
Comment (jieyilong) [reply rpc/query.go:805]: Yeah, let's refactor this into a util function so it can be called by both `GetEenpByHeight()` and `GetEenpStakeByHeight()`.

**Pull request merged**